src/gitpod.ts

```diff
@@ -0,0 +1,236 @@
+import { Component } from './component';
+import { Project } from './project';
+import { YamlFile } from './yaml';
+
+/**
+  * Relative path of Gitpod file to generate
+  */
+const GITPOD_FILE = '.gitpod.yml';
+
+/**
+ * https://www.gitpod.io/docs/configuration/
+ * https://www.gitpod.io/docs/config-start-tasks/
+ */
+
+/**
+ * If the standard Docker image provided by Gitpod does not include the tools
+ * you need for your project, you can provide a custom Docker image OR Dockerfile.
+ * https://hub.docker.com/r/gitpod/workspace-full/ is the default Gitpod image
+ */
+export interface GitpodDocker {
+
+  /**
+   *
+   * A publicly available image to use
+   *
+   * @default "jsii/superchain"
+   */
+  readonly image?: string;
+
+  /**
+   *
+   * @default
+   *
+   * @example
+   * {
+   *   file: '.gitpod.Docker',
+   * }
+   */
+  readonly file?: string;
+}
+
+/**
+  * Configure how the terminal should be opened relative to the previous task.
+  */
+export enum GitpodOpenMode {
+  /**
+   * Opens in the same tab group right after the previous tab
+   */
+  TAB_AFTER = 'tab-after',
+  /**
+   * Opens in the same tab group left before the previous tab
+   */
+  TAB_BEFORE = 'tab-before',
+  /**
+   * Splits and adds the terminal to the right
+   */
+  SPLIT_RIGHT = 'split-right',
+  /**
+   * Splits and adds the terminal to the left
+   */
+  SPLIT_LEFT = 'split-left',
+  /**
+   * Splits and adds the terminal to the tops
+   */
+  SPLIT_TOP = 'split-top',
+  /**
+   * Splits and adds the terminal to the bottom
+   */
+  SPLIT_BOTTOM = 'split-bottom'
+}
+
+/**
+ * Configure where in the IDE the terminal should be opened.
+ */
+export enum GitpodOpenIn {
+  /**
+   * the bottom panel (default)
+   */
+  BOTTOM = 'bottom',
+  /**
+   * the left panel
+   */
+  LEFT = 'left',
+  /**
+   * the right panel
+   */
+  RIGHT = 'right',
+  /**
+   * the main editor area
+   */
+  MAIN = 'main'
+}
+
+/**
+ * Configure options for a task to be run when opening a Gitpod
+ * workspace (e.g. running tests, or starting a dev server).
+ *
+ * Start Mode	        | Execution
+ * Fresh Workspace	  | before && init && command
+ * Restart Workspace	| before && command
+ * Snapshot	          | before && command
+ * Prebuild	          | before && init && prebuild
+*/
+export interface GitpodTask {
+
+  /**
+   * Required. What to run
+   */
+  readonly command: string;
+
+  /**
+   * A name for this
+   * @default
+   */
+  readonly name?: string;
+
+  /**
+   * You can configure how the terminal should be opened relative to the previous task.
+   * @default
```

Comment:
I found some more default information mentioned in https://github.com/gitpod-io/gitpod/blob/master/components/dashboard/public/schemas/gitpod-schema.json
```suggestion
   * @default GitpodOpenMode.TAB_AFTER
```

Reply:
done